main.tf

```
@@ -9,6 +9,11 @@ locals {
 
   create_tunner_with_internal_cidr_only = local.internal_cidr_provided && local.preshared_key_not_provided
   create_tunner_with_preshared_key_only = local.internal_cidr_not_provided && local.preshared_key_provided
+
+  connect_to_transit_gateway = var.transit_gateway_id != null ? true : false
+  connection_identifier = "${!local.connect_to_transit_gateway ? "VPC ${var.vpc_id}" : "TGW ${var.transit_gateway_id}"}"
```

No need to wrap this in `"${...}`

```suggestion
  connection_identifier = !local.connect_to_transit_gateway ? "VPC ${var.vpc_id}" : "TGW ${var.transit_gateway_id}"
```